### ISSUE TITLE
Safari & actions on native elements: Function statements must have a name.

### ISSUE THREAD

**Issue by rlivsey** [emberjs/ember.js#11640]
TLDR: http://jsbin.com/xuquqa/edit?html,js,console,output

I've seen a few demos doing the rounds of how awesome the new actions helpers are in combination with native elements, eg:

```handlebars
  <select onchange={{action "selectChanged" value="target.value"}}>
    {{#each items as |item|}}
      <option>...</option>
    {{/each}}
  </select>
```

This works great in Chrome, but in Safari this results in an error when the event is triggered:

```
SyntaxError: Function statements must have a name. (line 1)
```

[The JSBin](http://jsbin.com/xuquqa/edit?html,js,console,output) has examples of input and select both of which raise the same error.

**Comment by rwjblue** [comment]
I believe that this is a duplicate, and that we are unlikely to fix (unless someone knows what the solution to this cryptic error is off hand), as we intend to submit another RFC for something that I have been calling 'kabab actions'. The idea is to enable per element / event handling like your example above, but still use our standard event dispatcher (this avoids adding a ton of event listeners).

**Comment by rlivsey** [comment]
Cool, I couldn't find a duplicate issue but must have missed it - I'll close this one.

I wasn't sure it was intended to be supported, seemed like a "happy" accident, but as I'd seen all the demos knocking about I thought I'd give it a go.

Makes sense to go through the dispatcher as opposed to going back to the 90's and attaching handlers all over the shop.

**Comment by rwjblue** [comment]
No worries, I could be wrong about the dupe (SO MANY EMAILS :smiley_cat: ).

**Comment by stefanpenner** [comment]
lets keep this open until a duplicate is linked.

**Comment by stefanpenner** [comment]
I believe this is a bug in Safari. Which results in our detection code being thrown off.

@rwjblue this is directly related to my email yesterday.

**Comment by stefanpenner** [comment]
let me confirm

**Comment by cibernox** [comment]
This could have been terrible news for me. Glad I didn't enabled a new feature.

**Comment by stefanpenner** [comment]
yes this is related. I guess I need to role the fix sooner rather then later.

**Comment by raytiley** [comment]
Duplicate is #11604 I believe, although that one is currently closed.

**Comment by stefanpenner** [comment]
I have no idea why the other one is closed. The intent of HTML's props first, attrs later approach is the highest possible fidelity with the underlying DOM or WebComponent nodes.

**Comment by raytiley** [comment]
> I am about 5 failing tests away from what appears like a much better solution.

When are you not 5 failing tests away from a solution?

**Comment by stefanpenner** [comment]
right now. 0 failing tests, now just waiting on other things....

**Comment by rlivsey** [comment]
@raytiley ah that's why I didn't find it, thanks!

**Comment by stefanpenner** [comment]
https://github.com/tildeio/htmlbars/pull/374

**Comment by rwjblue** [comment]
Fixed in 1.13.3.

**Comment by stefanpenner** [comment]
thanks for releasing :)
